client/payment-methods/delete-modal.tsx

```diff
@@ -12,13 +13,22 @@ import interpolateComponents from 'interpolate-components';
 import PaymentDeleteIllustration from '../components/payment-delete-illustration';
 import ConfirmationModal from '../components/confirmation-modal';
 
+interface ConfirmPaymentMethodDeleteModalInput {
+	id: string;
+	label: string;
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	Icon: () => JSX.Element;
+	onConfirm: () => void;
+	onCancel: () => void;
+}
+
 const ConfirmPaymentMethodDeleteModal = ( {
 	id,
 	label,
 	Icon,
 	onConfirm,
 	onCancel,
-} ) => {
+}: ConfirmPaymentMethodDeleteModalInput ): JSX.Element => {
```

Comment:
Also: inline.

Reply:
Done. Is there a convention on when we should use `inline` vs a separate interface?

Reply:
Reusable interfaces are typically separate interfaces, others tend to be inline definitions.